### ISSUE TITLE
Error occurs while compiling

### ISSUE THREAD

**Issue by dorpauli** [lolepezy/rpki-prover#41]
Hello,

i wanted to update my instance of rpki-prover to the latest changes in this repository. I used the same environment as one month before, but a failure occurs. Should I use a completely new installation for the compilation? Does the last compilation affect this new compilation?

Here's the error output of stack:
```
[59 of 60] Compiling RPKI.TopDown
/home/ubuntu/rpki-prover/src/RPKI/TopDown.hs:63:1: error:
    Could not find module ‘RPKI.Fetch’
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
   |
63 | import           RPKI.Fetch
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^


Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/AppContext.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/AppMonad.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/AppState.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/CommonTypes.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Config.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Domain.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Http/Api.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Http/HttpServer.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Http/Messages.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Http/Types.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Http/UI.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Logging.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Metrics.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Orphans/Generics.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Orphans/Serialise.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Parallel.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Parse/Internal/Cert.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Parse/Internal/Common.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Parse/Internal/CRL.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Parse/Internal/GBR.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Parse/Internal/MFT.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Parse/Internal/ROA.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Parse/Internal/SignedObject.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Parse/Parse.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Reporting.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Repository.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Resources/IntervalSet.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Resources/Resources.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Resources/Types.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/RRDP/Http.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/RRDP/Parse.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/RRDP/RrdpFetch.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/RRDP/Types.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Rsync.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/RTR/Pdus.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/RTR/RtrServer.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/RTR/RtrState.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/RTR/Types.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Store/AppLmdbStorage.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Store/AppStorage.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Store/Base/InMemory.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Store/Base/LMDB.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Store/Base/Map.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Store/Base/MultiMap.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Store/Base/Storable.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Store/Base/Storage.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Store/Database.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Store/MakeInMemory.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Store/MakeLmdb.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Store/Repository.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Store/Sequence.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/TAL.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Time.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Util.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Validation/Crypto.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Validation/ObjectValidation.hi Decoding failure: Invalid magic: e49ceb0f

Warning: Failed to decode module interface: /home/ubuntu/rpki-prover/.stack-work/dist/aarch64-linux/Cabal-3.2.1.0/build/RPKI/Validation/ResourceValidation.hi Decoding failure: Invalid magic: e49ceb0f
Completed 208 action(s).

--  While building package rpki-prover-0.1.0.0 using:
      /home/ubuntu/.stack/setup-exe-cache/aarch64-linux/Cabal-simple_mPHDZzAJ_3.2.1.0_ghc-8.10.4 --builddir=.stack-work/dist/aarch64-linux/Cabal-3.2.1.0 build lib:rpki-prover exe:rpki-prover --ghc-options " -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
```

**Comment by lolepezy** [comment]
It probably has to do with some confusions of GHC/Stack versions. It's switched to GHC 8.10.4 not too long ago, so I would recommend to do at least `stack clean` first. If the problem still persists, it's would help deleting `.stack-work` directory.

**Comment by lolepezy** [comment]
Another option is that you are using `stack` version that doesn't support  GHC 8.10.4. In this case, just do `stack upgrade`.

**Comment by lolepezy** [comment]
Ah, sorry I'm wrong, it's an actual problem in the code, not the environment.
Not sure how, some unfinished merge artifacts leaked into the `master` branch. 
All fixed now.

Also, if you tested it about a month ago, I would recommend deleting the content of `~/.rpki/cache/` 
directory, it may be not compatible with the current master anymore.

**Comment by dorpauli** [comment]
Ok thanks, now it compiles til the end and the binary is in `~/.local/bin/`. 

The only confusing thing is, that the warnings for the decoding issues are still there. Maybe it's because of GHC and stack itself. The problem is, that i build on an arm based machine and this is currently not supported. So i did a workaround with ubuntu 18 and an older stack version (2.1.3), because this was working for arm in the past. 

I don't remember, if the warning were there the last time i compiled, but the binary from then works fine.

**Comment by lolepezy** [comment]
As far as I can tell, these warning are pretty harmless (some stack functionality won't work, but the basic one works fine). It is caused by Stack trying to parse the interface files (*.hi) for some reason and failing to do it for newer GHC versions.

Does the binary work now?

**Comment by dorpauli** [comment]
Yes it works! I really like the new UI.

I also followed your advice to run stack upgrade and it surprisingly upgraded to the current stack version by compiling itself. This did not happen a month ago. It also uses ghc 8.10.4 now. So i have a working stack 2.5.1.1 aarch64 now. I don't understand why i found nothing like this on the internet. 😄 

I compiled the rpki-prover with this again, but it's the same output as with 2.1.3. 😉
